util/suffix.c

```diff
@@ -65,36 +68,87 @@ const char *suffix_si_get(double *value)
 	return suffix;
 }
 
-uint64_t suffix_si_parse(const char *value, bool *suffixed)
+int suffix_si_parse(const char *str, char **endptr, uint64_t *val)
 {
-	char *suffix;
-	double ret;
-	struct si_suffix *s;
-
-	errno = 0;
-	ret = strtod(value, &suffix);
-	if (errno)
+	unsigned long long num, frac;
+	char *sep, *tmp;
+	int frac_len, len, i;
+
+	num = strtoull(str, endptr, 0);
+	if (str == *endptr ||
+	    ((num == ULLONG_MAX) && errno == ERANGE))
+		return -EINVAL;
+
+	/* simple number, no decimal point not suffix */
+	if (*endptr[0] == '\0') {
```

Comment:
This also should be fixed as same with other casting `endptr` parts. (By the way it seems that it is same with the case without the casting for the pointer of pointer as `(*endptr)[n]` and `*endptr[n]` are same but not sure..)
```diff
- *endptr[0]
+ (*endptr)[0]
```